lib/src/async_provider.dart

```diff
@@ -127,42 +127,45 @@ class StreamProvider<T> extends DeferredInheritedProvider<Stream<T>, T> {
 DeferredStartListening<Future<T>, T> _futureStartListening<T>({
   T initialData,
   ErrorBuilder<T> catchError,
-}) {
-  return (e, setState, controller, __) {
-    if (!e.hasValue) {
-      setState(initialData);
-    }
+}) =>
```

Comment:
I don't like this change.
If that's caused by a warning, we should disable the said warning.